pkg/api/server/v1alpha2/server.go

```diff
@@ -47,3 +51,17 @@ func New(db *gorm.DB) (*Server, error) {
 
 	return srv, nil
 }
+
+// GetResultByName is the helper function to get a Result by its parent and name
+func GetResultByName(gdb *gorm.DB, name string) (*pb.Result, error) {
```

Comment:
This near-identical to GetResult, but is missing the error handling that is in GetResult. We should refactor these together so that they share the same impl but allow for different transaction types to be used. (we should also move this into result.go with the rest of the CRUD operations).

Reply:
Ok!